python/tank_vendor/shotgun_deploy/descriptor.py

```diff
@@ -0,0 +1,286 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import copy
+
+from . import paths
+from .descriptor_io import create_io_descriptor
```

Comment:
minor: `descriptor_io` vs. `io_descriptor`. Wondering if there should be ore consistency by calling the file `io_descriptor` since that seems to be how these are referred to (`IODescriptor*` classes etc) throughout the code. I see the form is mirrored with `bundle`, `config`, and `core` descriptors too. Just something to consider.